### PR TITLE
Add new flags and make old persistence flags become new ones' aliases

### PR DESCRIPTION
**Pull request by Zongrenfeng**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description
Add new flags, and make the old ones become new ones' alias

### CODE REVIEW

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 1]
```diff
@@ -17,17 +17,23 @@
...
+    public static final int CRITICAL = 0x02;
+
+    public static final int PERSISTENCE_CRITICAL = CRITICAL;
 
     /**
      * Default combination of flags.
      */
-    public static final int DEFAULTS = PERSISTENCE_NORMAL;
+    public static final int DEFAULT_FLAGS = NORMAL;
+
```
Comment (Jamminroot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]:
```suggestion
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 2]
```diff
@@ -17,17 +17,23 @@
     /**
      * An event can be lost due to low bandwidth or disk space constraints.
      */
-    public static final int PERSISTENCE_NORMAL = 0x01;
+    public static final int NORMAL = 0x01;
+
```
Comment (Jamminroot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]:
```suggestion
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 3]
```diff
@@ -17,17 +17,23 @@
     /**
      * An event can be lost due to low bandwidth or disk space constraints.
      */
-    public static final int PERSISTENCE_NORMAL = 0x01;
```
Comment (MatkovIvan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: Keep it

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java (hunk 1 above)
Comment (Zongrenfeng) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: Removed empty line.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 4]
```diff
@@ -17,17 +17,23 @@
...
+    public static final int NORMAL = 0x01;
+
+    public static final int PERSISTENCE_NORMAL = NORMAL;
 
     /**
      * Used for events that should be prioritized over non-critical events.
      */
-    public static final int PERSISTENCE_CRITICAL = 0x02;
+    public static final int CRITICAL = 0x02;
+
```
[the remark this answers is not in the document]
Comment (Zongrenfeng) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: Removed empty line.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java (hunk 2 above)
Comment (Zongrenfeng) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: Removed empty line.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java (hunk 3 above)
Comment (Zongrenfeng) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: Updated.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 5]
```diff
@@ -18,16 +18,19 @@
      * An event can be lost due to low bandwidth or disk space constraints.
      */
     public static final int PERSISTENCE_NORMAL = 0x01;
+    public static final int NORMAL = PERSISTENCE_NORMAL;
```
Comment (MatkovIvan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: Add separate comments for new constants

Comment (annakocheshkova) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: And mark old ones as deprecated

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 6]
```diff
@@ -19,15 +19,30 @@
...
 
+    /**
+     * A flag as an alias of PERSISTENCE_NORMAL.
+     */
+    public static final int NORMAL = PERSISTENCE_NORMAL;
+
     /**
      * Used for events that should be prioritized over non-critical events.
      */
     public static final int PERSISTENCE_CRITICAL = 0x02;
```
Comment (MatkovIvan) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: Please mark old constants as deprecated

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 7]
```diff
@@ -19,15 +19,30 @@
...
+    /**
+     * Flag for all default flags including latency and future flags
+     */
+    public static final int DEFAULT_FLAGS = NORMAL;
+
     /**
      * Default combination of flags.
      */
-    public static final int DEFAULTS = PERSISTENCE_NORMAL;
+    public static final int DEFAULTS = DEFAULT_FLAGS;
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]: This was an old mistake in a reused specification document, please remove `DEFAULT_FLAGS`, we still use only `DEFAULTS` like before.

**Comment by annakocheshkova** [comment]
We should update this PR once all the other feature PRs are merged and merge after them.

File: sdk/appcenter-push/src/test/java/com/microsoft/appcenter/push/PushTest.java [hunk 8]
```diff
@@ -681,7 +681,7 @@ public void registerWithoutFirebaseButUseGoogleServicesStringForSenderId() {
         when(FirebaseInstanceId.getInstance()).thenThrow(exception);
         when(mContext.getPackageName()).thenReturn("com.contoso");
         Resources resources = mock(Resources.class);
-        when(resources.getIdentifier("gcm_defaultSenderId", "string", "com.contoso")).thenReturn(42);
+        when(resources.getIdentifier("gcm_DEFAULTSenderId", "string", "com.contoso")).thenReturn(42);
```
Comment (jaelim-ms) [inline sdk/appcenter-push/src/test/java/com/microsoft/appcenter/push/PushTest.java]: Is this change related to the PR?

Comment (Zongrenfeng) [reply sdk/appcenter-push/src/test/java/com/microsoft/appcenter/push/PushTest.java]: sorry, my mistaken. thanks~

**Comment by MatkovIvan** [comment]
@Zongrenfeng Please resolve conflicts with the base branch

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 9]
```diff
@@ -15,19 +15,31 @@
 public final class Flags {
 
     /**
-     * An event can be lost due to low bandwidth or disk space constraints.
+     * @deprecated Use of flag PERSISTENCE_NORMAL has been deprecated, suggest to use new flag NORMAL as replacement.
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]:
```suggestion
     * @deprecated This flag has been deprecated, use {@link #NORMAL} instead.
```

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java [hunk 10]
```diff
@@ -15,19 +15,31 @@
...
     public static final int PERSISTENCE_NORMAL = 0x01;
 
     /**
-     * Used for events that should be prioritized over non-critical events.
+     * An event can be lost due to low bandwidth or disk space constraints.
+     */
+    public static final int NORMAL = PERSISTENCE_NORMAL;
+
+    /**
+     * @deprecated Use of flag PERSISTENCE_CRITICAL has been deprecated, suggest to use new flag CRITICAL as replacement.
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/Flags.java]:
```suggestion
     * @deprecated This flag has been deprecated, use {@link #CRITICAL} instead.
```
